crates/bevy_color/src/color_ops.rs

```diff
@@ -61,3 +61,18 @@ pub trait Alpha: Sized {
         self.alpha() >= 1.0
     }
 }
+
+/// Trait with methods for asserting a colorspace is within bounds.
+pub trait ClampColor: Sized {
+    /// Return a new version of this color clamped, with all fields in bounds.
+    fn clamp(&self) -> Self;
+
+    /// Changes all the fields of this color to assert they are within bounds.
+    fn clamp_self(&mut self);
+}
+
+/// Trait for verifying all the fields of a color is in the defined range.
+pub trait IsWithinBounds: Sized {
```

Comment:
IMO the traits can be combined to a single `ColorBounds` trait that implements both the checking and the clamping. Or maybe `Clamp` should extend `WithinBounds`.

Reply:
In the beginning i was thinking this too. But I combined then for now, I can change if it's a netter choice.

Reply:
Yeah, I like this better as one trait.